crypto/x509/x509_vfy.c

```diff
@@ -899,6 +899,8 @@ static int check_trust(X509_STORE_CTX *ctx, int num_untrusted)
      */
     for (i = num_untrusted; i < num; i++) {
         x = sk_X509_value(ctx->chain, i);
+        if (x == NULL)
+            return -1;
```

Comment:
This doesn't seem quite right.   check_trust returns one of (X509_TRUST_TRUSTED, X509_TRUST_REJECTED, X509_TRUST_UNTRUSTED), which the two call sites test for.  I think this should probably be returning X509_TRUST_UNTRUSTED

Reply:
It returns -1 on internal error. IMO this is a _cannot happen_ error so -1 return is OK. The condition should probably be an assert type: `!ossl_assert(x != NULL)` that is.

Reply:
The comments say that, but this appears to be the first that we are actually using it, and call sites for check_trust and callers don't appear to be prepared to handle this.  For example build_chain will return -1 if check_trust does, but X509_build_chain only checks build_chain for 0, not -1 in the error case.

I can understand using -1 here as an internal error code since we have comments that indicate thats what it does, but the callers aren't currently prepared to handle it.

I agree this should never happen, and be an assert, but in general, if we want to start returning -1 here, the call sites need to be altered to handle that

Reply:
True, I am afraid though that such PR would no longer be acceptable with CLA: trivial.